signal_handler.c

```diff
@@ -175,26 +200,42 @@ PHP_FUNCTION(attach_signal)
 
 	/* Check if given parameter is callable */
 	#if ZEND_MODULE_API_NO <= 20060613
-		if (!zend_is_callable(handle, 0, &func_name)) {
+	if (!zend_is_callable(handle, 0, func_name)) {
+	#elseif PHP_MAJOR_VERSION >= 7
+	if (!zend_is_callable(handle, 0, &func_name)) {
```

Comment:
You can actually combine the PHP_MAJOR_VERSION >= 7 with the else block below since TSRMLS_CC evaluates to nothing in PHP7.  So just get rid of lines 204 and 205 and it'll be fine *and* more readable.